x-pack/solutions/security/plugins/security_solution/server/lib/trial_companion/routes/nba_routes.ts

```diff
@@ -0,0 +1,224 @@
+  IKibanaResponse,
+  KibanaResponseFactory,
+  Logger,
+  RequestHandler,
+} from '@kbn/core/server';
+import { transformError } from '@kbn/securitysolution-es-utils';
+import { buildSiemResponse } from '@kbn/lists-plugin/server/routes';
+import type { TypeOf } from '@kbn/config-schema';
+import { schema } from '@kbn/config-schema';
+import { type Either, left, match, right } from 'fp-ts/Either';
+import { pipe } from 'fp-ts/function';
+import type { TrialCompanionUserNBAService } from '../services/trial_companion_user_nba_service.types';
+import { TRIAL_COMPANION_NBA_URL } from '../../../../common/trial_companion/constants';
+import type { SecuritySolutionRequestHandlerContext } from '../../../types';
+import { TrialCompanionUserNBAServiceImpl } from '../services/trial_companion_user_nba_service';
+import type { TrialCompanionRoutesDeps } from '../types';
+
+// TODO: think about dependency injection of saved objects (internal) client. Currently, we get it from context.core.savedObjects
+// TODO: if we need internal client - add user milestone service in x-pack/solutions/security/plugins/security_solution/server/request_context_factory.ts
+// TODO: and get it from request context
+
+interface NBAContext {
+  nbaService: TrialCompanionUserNBAService;
+  username: string;
+}
+
+const NBARouteSchema = {
+  body: schema.object({
+    milestoneId: schema.number(), // TODO: oneOf plus iterate over ALL_NBA keys
+  }),
+};
+
+type NBASeenRouteRequestBody = TypeOf<typeof NBARouteSchema.body>;
+
+const notFound: RequestHandler<never, never, never, SecuritySolutionRequestHandlerContext> = async (
+  context,
+  request,
+  response
+) => {
+  return response.notFound({
+    body: 'disabled',
+  });
+};
+
+export const registerGetNBARoute = ({ router, logger, enabled }: TrialCompanionRoutesDeps) => {
+  router.versioned
+    .get({
+      path: TRIAL_COMPANION_NBA_URL,
+      access: 'internal',
+      summary: 'Get Trial Companion NBA for a user',
+      options: {
+        tags: ['api'],
+      },
+      security: {
+        authz: {
+          requiredPrivileges: ['securitySolution'],
+        },
+      },
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: false, // TODO: help needed here - do we need to validate something?
+      },
+      enabled ? getCurrentNBAForUser(logger) : notFound
+    );
+};
+
+export const registerPostNBASeenRoute = ({ router, logger, enabled }: TrialCompanionRoutesDeps) => {
+  router.versioned
+    .post({
+      path: TRIAL_COMPANION_NBA_URL,
+      access: 'internal',
+      summary: 'Save Trial Companion NBA seen action (aka dismiss)',
+      options: {
+        tags: ['api'],
+      },
+      security: {
+        authz: {
+          requiredPrivileges: ['securitySolution'],
+        },
+      },
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: NBARouteSchema,
+        },
+      },
+      enabled ? postNBAUserSeen(logger) : notFound
```

Comment:
Isn't it more explicit to just not register the routes at all when the FF is disabled? It simplifies route definition (I mean, why should the route itself be aware of the FF?).

Reply:
That is a good point. Originally I wanted to avoid a call to register APIs when our experimental feature is disabled. Then I have read about feature flag service, which recommends to register in any case. 

If you think that there is no way to change experimental features while kibana running? If so, I would change the code (it will be simpler).